src/DynamoCore/Extensions/ExtensionManager.cs

```diff
@@ -154,6 +167,14 @@ public IEnumerable<IExtension> Extensions
             get { return extensions; }
         }
 
+        /// <summary>
+        /// Returns the collection of registered extensions implementing IExtensionStorageAccess
+        /// </summary>
+        public IEnumerable<IExtensionStorageAccess> StorageAccessesExtensions
+        {
+            get { return storageAccessExtensions; }
```

Comment:
why not use an autoproperty?

Reply:
No reason not to, just wanted to follow same pattern that is used for the `extensions` collection.

Reply:
If theres no reason - use the autoproperty - less code, same effect. I'm not sure if the extensions backing field is accessed directly somewhere for some reason?

Reply:
Actually when thinking about it, we dont want to enable adding new `IExtensionStorageAccess` anywhere else than from the managers so therefor it makes sense to return an IEnumerable like this.

If we have it as an autoproperty we would have to return the actual collection which would enable adding IExtensionStorageAccess to the collection from other classes. So seems to me this is the way to do it with less code. Unless there is some other way to achieve that which im not familiar with?

Reply:
can you explain why a property without a backing field would force you to return the actual collection?

Reply:
Sorry if im misunderstanding this, but we cant add directly to an `IEnumerable`.

So the reason for having a backing field is that we in the `ExtensionManager` can add new extensions to the `List<IExtensionStorageAccess>` but we only expose the `IEnumerable` to the outside. 

I think im struggling to see how we can do it with less code than
```csharp
private List<IExtensionStorageAccess> storageAccessExtensions = new List<IExtensionStorageAccess>(); 

public IEnumerable<IExtensionStorageAccess> StorageAccessExtensions { get { return storageAccessExtensions; }}
```

I guess we could use an `ReadOnlyCollection` but havent really looked into them before, and not sure it would minimize the amount code very much.

Reply:
I *think* you can write this:

```
public IEnumerable<IExtensionStorageAccess> StorageAccessExtensions {get;} = new List<IExtensionStorageAccess>();

```

or 

```

some internal method(){

StorageAccessExtensions = new List<IExtensionStorageAccess>();

}
```

`IEnumerable<T>` is just an interface, the concrete type can still be a list. BUT I see that we do a bunch of internal stuff to that backing field which needs to be an `ICollection` and I *think* we'd have to do a bunch of funny casts for no reason so this looks good to me. 👍